Feature/#30 feat timetable name limit

BEEEAM-J opened the pull request:
## 📌 PR 요약

🌱 작업한 내용
- 시간표 이름 20자 제한 기능 추가
-  CchAppBarWithTitle "뒤로 가기" 버튼 영역 수정

🌱 PR 포인트
- 시간표 이름 20자 제한 기능 -> 시간표 이름 입력 화면, 시간표 수정 화면 적용

## 📸 스크린샷
|스크린샷|
|:--:|
|![1](https://github.com/user-attachments/assets/c4912e3b-9903-4a40-a69e-14327514daf1)|
|![2](https://github.com/user-attachments/assets/60961551-bfc3-4c40-88f4-3943b2835233)|


## 📮 관련 이슈
- Resolved: #30 

### RCA 룰을 사용하여 코드 리뷰를 해주세요
`R (Request Changes)` : 적극적으로 반영을 고려해주세요   
`C (Comment)` : 웬만하면 반영해주세요   
`A (Approve)` : 반영해도 좋고, 넘어가도 좋습니다. 사소한 의견입니다.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **신기능**
  - 시간표 이름이 20자를 초과하면 텍스트 필드에 에러 상태와 안내 문구를 표시
  - 텍스트 필드가 에러 상태 표시(isError)를 지원

- **버그 수정**
  - 뒤로가기 버튼의 터치 동작 일관성 개선
  - 시간표 이름 입력/편집 화면의 버튼 활성화 로직이 검증 규칙을 따르도록 수정
  - 시간표 목록의 빈 상태 문구와 배치(패딩) 조정으로 가시성 향상
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

BEEEAM-J committed:
Fix: appBar 뒤로가기 버튼 클릭 영역 확대

- `Image`에 `cchClickable` Modifier를 적용하여 클릭 영역을 명확히 함

BEEEAM-J committed:
refactor: 시간표 리스트 문구 수정

- 시간표 리스트가 비어있을 때 표시되는 문구를 "등록된 시간표가 없어요!"로 변경
- 문구의 상단 패딩 값을 150.dp에서 324.dp로 변경

BEEEAM-J committed:
feat: 시간표 이름 입력 유효성 검사 추가

- 시간표 이름이 20자를 초과하는 경우 에러 상태로 표시
- 텍스트 필드에 `isError` 파라미터 추가하여 에러 상태 시 테두리 색상 변경
- 에러 메시지를 표시하여 사용자에게 입력 제한을 알림

BEEEAM-J committed:
feat: 시간표 이름 입력 유효성 검사 추가

- 시간표 이름이 20자를 초과하는 경우 에러 상태로 표시
- 텍스트 필드에 `isError` 파라미터 추가하여 에러 상태 시 테두리 색상 변경
- 에러 메시지를 표시하여 사용자에게 입력 제한을 알림

lluke0 commented on composeApp/src/commonMain/kotlin/com/chukchukhaksa/mobile/presentation/timetable/timetablenameinput/TimetableNameInputContract.kt:
```diff
@@ -9,7 +9,7 @@ data class TimetableNameInputState(
     val semester: Semester = Semester("", "")
 ) {
-    val buttonEnabled = name.isNotEmpty()
+    val buttonEnabled = name.isNotEmpty() && name.length <= 20
```
이건 반영해주시죠 @BEEEAM-J

BEEEAM-J replied:
코멘트 내용 반영했습니다~ @jinukeu

lluke0 replied:
넵 확인이용 바로 머지 가시죠! @BEEEAM-J

BEEEAM-J committed:
refactor: 시간표 이름 글자 수 제한 로직 변경

- 시간표 이름 글자 수 제한 상수 `TIMETABLE_NAME_LIMIT` 정의
- `checkOverTimetableNameLimit`: 시간표 이름이 글자 수 제한을 초과하는지 확인
- `checkTimetableNameRule`: 시간표 이름이 유효한지(공백이 아니고 글자 수 제한 이내) 확인
